src/app/components/Core/Selector.tsx

```diff
@@ -0,0 +1,208 @@
+import React from 'react';
+
+import Select, {
+  FocusEventHandler,
+  InputActionMeta,
+  OptionProps,
+  OptionTypeBase,
+  components as ReactSelectComponents,
+  SelectComponentsConfig,
+  Styles,
+  ValueContainerProps
+} from 'react-select';
+import styled from 'styled-components';
+
+import expand from '@app/assets/icons/expand.svg';
+import { Image } from '@app/components';
+
+interface SelectorProps<T extends OptionTypeBase> {
+  options: T[];
+  // We prefer controlled components so `value` is required prop. When it is `null`, React-Select will display the placeholder
+  value: T | null | undefined;
+  disabled?: boolean;
+  placeholder?: string;
+  searchable?: boolean;
+  clearable?: boolean;
+  name?: string;
+  optionComponent: React.ComponentType<OptionProps<T>>;
+  valueComponent?: React.ComponentClass<{ value: T }> | React.StatelessComponent<{ value: T }>;
+  inputId?: string;
+  inputValue?: string;
+  onCloseResetsInput?: boolean;
+  onBlurResetsInput?: boolean;
+  onBlur?: FocusEventHandler;
+  optionDivider?: boolean;
+  isClearable?: boolean;
+  components?: SelectComponentsConfig<T>;
+  getOptionLabel?(option: T): string;
+  onChange?(option: T): void;
+  onInputChange?(newValue: string, actionMeta: InputActionMeta): void;
+  onInputKeyDown?(e: React.KeyboardEvent<HTMLElement>): void;
+}
+
+// Fixes weird placement issues for react-select
+const Wrapper = styled('div')`
+  cursor: pointer;
+  &:hover {
+    cursor: default;
+  }
+`;
+
+const OptionWrapper = styled.div`
+  ${(p: { optionDivider: boolean }) =>
+    p.optionDivider
+      ? `
+    border-bottom: 1px solid #e5ecf3;
+    &:last-of-type {
+      border-bottom: none
+    }
+  `
+      : ''}
+
+  &:hover {
+    background-color: #f7f7f7;
+  }
+`;
+
+export const DropdownIndicatorWrapper = ReactSelectComponents.DropdownIndicator;
+export const ClearIndicatorWrapper = ReactSelectComponents.ClearIndicator;
+
+const getValueContainer: <T = any>(
+  props: ValueContainerProps<T> & OptionProps<T>,
+  ValueComponent?: React.ComponentType<{ value: T }>
+) => any = (props, ValueComponent) => {
+  const {
+    hasValue,
+    getValue,
+    selectProps: { inputValue }
+  } = props;
+
+  if (!hasValue || !ValueComponent) {
+    return <ReactSelectComponents.ValueContainer {...props} />;
+  }
+
+  const [data] = getValue() as any[];
+  const { children } = props;
+
+  return (
+    <ReactSelectComponents.ValueContainer {...props}>
+      {React.Children.map(children, (child: JSX.Element) =>
+        child && [ReactSelectComponents.SingleValue].indexOf(child.type) === -1 ? child : null
+      )}
+      {!inputValue && <ValueComponent value={data} />}
+    </ReactSelectComponents.ValueContainer>
+  );
+};
```

Comment:
This function can be typed a lot better, e.g.:
```suggestion
const getValueContainer = <T extends unknown = unknown>(
  props: ValueContainerProps<T> & OptionProps<T>,
  ValueComponent?: ComponentType<{ value: T }>
): ReactElement => {
  const {
    hasValue,
    getValue,
    selectProps: { inputValue }
  } = props;

  if (!hasValue || !ValueComponent) {
    return <ReactSelectComponents.ValueContainer {...props} />;
  }

  const [data] = getValue() as T[];
  const { children } = props;

  return (
    <ReactSelectComponents.ValueContainer {...props}>
      {React.Children.map(children, (child: JSX.Element) =>
        child && [ReactSelectComponents.SingleValue].indexOf(child.type) === -1 ? child : null
      )}
      {!inputValue && <ValueComponent value={data} />}
    </ReactSelectComponents.ValueContainer>
  );
};
```